presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -169,6 +169,14 @@ private void validateShowStats(ShowStats node)
                 check(query.getQueryBody() instanceof QuerySpecification, node, "Only table and simple table subquery can be passed as argument to SHOW STATS clause");
                 QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
 
+                Plan plan = queryExplainer.get().getLogicalPlan(session, new Query(Optional.empty(), querySpecification, Optional.empty(), Optional.empty()), parameters);
```

Comment:
This logic is duplicated in `getConstraint`. Consider refactoring this code a bit to avoid that duplication. Specifically, `validateShowStats` is needed only if `node.getRelation() instanceof TableSubquery`. So, you could do something like:

```
if (node.getRelation() instanceof TableSubquery) {

    Plan plan = ...
    validate(...plan);
     ...getConstraint(...plan...)
}
```

Reply:
I initially tried this and pass plan around,  but I am not sure it is good to pass plan into validateShowStats, good to know "validateShowStats is needed for subquery".